assets/src/data/model/base-date-formatter.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * External imports
+ */
+import * as dateFormats from '@eventespresso/helpers';
+import { isEmpty } from 'lodash';
+
+/**
+ * Formats the date fields on provided entities.  Does not mutate original
+ * entities.
+ *
+ * @param { Array } entities  An array of entity objects
+ * @param { Array } entityDateFields  An array of field names that are date
+ *   fields.
+ * @param { string } format  The format to transform the date field values to.
+ * @param { boolean } local      Whether or not to convert the date field value
+ *   to the local timezone for the host.
+ * @return { Array }  Returns a new array of new entities with the date field
+ *   values formatted
+ */
+export const formatDatesOnEntities = (
+	entities = [],
+	entityDateFields = [],
+	format = dateFormats.DATE_TIME_FORMAT_ISO8601,
+	local = true,
+) => {
+	if ( isEmpty( entities ) || isEmpty( entityDateFields ) ) {
+		return entities;
+	}
+	const formattedEntities = [];
+	entities.forEach( ( entity ) => {
+		formattedEntities.push( formatDatesOnEntity(
+			entity,
+			entityDateFields,
+			format,
+			local,
+		) );
+	} );
+	return formattedEntities;
+};
+
+/**
+ * Formats the date fields on the provided entity.  Does not mutate original
+ * entity.
+ *
+ * @param { Object } entity  An entity
+ * @param { Array } entityDateFields  An array of field names that are date
+ *   fields.
+ * @param { string } format  The format to transform the date field values to.
+ * @param { boolean } local      Whether or not to convert the date field value
+ *   to the local timezone for the host.
+ * @return { Object }  Returns a new entity with the date field values formatted
+ */
+export const formatDatesOnEntity = (
+	entity = {},
+	entityDateFields = [],
+	format = dateFormats.DATE_TIME_FORMAT_ISO8601,
+	local = true,
+) => {
+	const newEntity = { ...entity };
+	entityDateFields.forEach( ( dateField ) => {
+		if ( newEntity[ dateField ] ) {
+			newEntity[ dateField ] = dateFormats.formatDateString(
+				newEntity[ dateField ],
+				format,
+				local,
+			);
+		}
+	} );
+	return newEntity;
+};
+
+/**
+ * Formats the date fields to mysql format on provided entities.  Does not
+ * mutate original entities.
+ *
+ * @param { Array } entities  An array of entity objects
+ * @param { Array } entityDateFields  An array of field names that are date
+ *   fields.
+ * @param { boolean } local      Whether or not to convert the date field value
+ *   to the local timezone for the host.
+ * @return { Array }  Returns a new array of new entities with the date field
+ *   values formatted
+ */
+export const datesToMysqlFormatOnEntities = (
```

Comment:
following the same pattern you have set already in `assets/src/data/helpers/datetime.js` (ie: `formatDateString()`, `formatMysqlDateString()`, `formatSiteDateString()`, etc), how about naming this `formatMysqlDatesOnEntities()` ?

or... 

if you are trying to be more descriptive, then rewrite the function name to be more inline with how people actually talk...
ie: you would never say "Hey, let's dates to MySQL format on the entities ok?"
but you might say something like "Hey, let's format the entities dates to MySQL ok?"
So that latter sentence might translate into a function name like: `formatEntitiesDatesToMysql()` which is far easier to understand the purpose of than `datesToMysqlFormatOnEntities()`